blocks/layout-grid/src/icons.js

```diff
@@ -5,16 +5,17 @@
 import { Path, SVG } from '@wordpress/components';
 
 export const GridIcon = ( props ) => {
+	const iconProps = {...props};
 	if ( props.size ) {
-		props.width = props.size;
-		props.height = props.size;
+		iconProps.width = props.size;
+		iconProps.height = props.size;
 	}
```

Comment:
When the size isn't set is it equivalent to having the width and height set to 0? If so we could simplify the code by leveraging the optional chaining operator and null coalescing operator and end up with (if zero isn't good we can think of another default value):
```
const iconProps = {
    width: props?.size ?? 0,
    height: props?.size ?? 0,
    ...props
};
```

Reply:
Omitting width and height would mean they get set to 'auto' (resolving `auto` is a whole can of worms but we don't need to worry about that). 

I don't think your suggestion is really significantly simpler. I'll change it if you disagree though.

Reply:
Given your explanation that omitting width and height would set them to 'auto', it indeed seems your implementation has a different behavior than the one I suggested. I was under the impression that the absence of these properties would default to '0', not 'auto'. Therefore, if 'auto' is the intended behavior when props.size is not provided, your current implementation seems appropriate.

As a general rule, small steps towards simpler code can aggregate into a significant simplification of our code base over time, that's why I usually insist on applying small or marginal simplifications.